monitor_other_boards.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+from datetime import datetime
+import subprocess
+
+import slack_reporter
+
+#                SSH target               name
+all_boards = (("odroid@odroid",         "odroid-C4"),
+              ("orangepi@oliviaorange", "orange pi 02"),
+              ("viam@viam",             "Up 4000"),
+              ("viam@orinnanodevkit2",  "Orin Nano"),
+              # The Orin AGX is unable to get on tailscale, and mDNS is not
+              # very reliable. but its IP address (almost?) never changes.
+              ("viam@10.1.8.37",        "Orin AGX"),
+              ("debian@bogglebean",     "Beaglebone AI-64"),
+              ("canary@pi5canary",      "rpi5"),
+              )
+
+
+def ensure_board_is_online(ssh_target, name):
+    today = datetime.today().strftime("%Y-%m-%d")
+    # This command SSHes into the target machine and counts how many times the
+    # canary analysis logfile contains a line with today's date and nothing
+    # else. If the analysis has run, there should be exactly 1 of these lines.
+    command = f"cat /var/log/canary_tests.log | grep '^{today}$' | wc -l"
+    subprocess_command = f"ssh '{ssh_target}' \"{command}\""
```

Comment:
How does this line know where the key is? We could either copy it into /root/.ssh/id_ed25519 on the monitor board, or put it in this directory and add a `-i id_ed25519` to this line. I'm unsure which I prefer: do we want to keep this line clean and straightforward, or do we want to keep the root user's environment clean and straightforward?

Reply:
If we put the private key in the local directory, anyone who can get into the board can read it, which would let them SSH into any other canary. If we put it in `/root/.ssh`, anyone on this board would need sudo access use it. but frankly, anyone who can get into the monitor already has sudo access on it and all the other canaries, too. Where we put it might be a distinction without a difference.

Reply:
I'll keep the private key in the local directory. That makes it easier to run/debug from my laptop, too.